### DISCUSSION TITLE
Is it useful to add data fetching function inside store?

### DISCUSSION THREAD

**Discussion by arsekil** [pmndrs/zustand#2602] [Q&A]
Hello,

I had problems with useState and useEffect, that it didn't update state effectively,so i decided to use Zustand for state management.

Is it logical to add the userdata-fetching function inside the store like this?

```
getUserData: async () => {
 const user = await FetchUser();
set({ user: user})
}
```

**Comment by dbritto-dev** [comment]
@andrejmoltok yeah, valid code

**Reply by arsekil** [reply]
Thank you for your response
